### PR TITLE
Added correlation Id checking and docs. 

### PR DESCRIPTION
**Pull request by parkeradam**
Auto-updates of 2020 to 2021 for copyright

### CODE REVIEW

File: app/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingController.scala [hunk 1]
```diff
@@ -42,17 +42,20 @@ abstract class PrivilegedCitizenMatchingController(
...
-      withJsonBody[CitizenMatchingRequest] { matchCitizen =>
-        citizenMatchingService.matchCitizen(matchCitizen) map { matchId =>
-          val selfLink = HalLink("self", s"/individuals/matching/")
-          val individualLink = HalLink(
-            "individual",
-            s"/individuals/matching/$matchId",
-            name = Option("GET"),
-            title = Option("Individual Details"))
-          Ok(links(selfLink, individualLink))
+      withCorrelationId { () =>
```
Comment (owen-opencast) [inline app/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingController.scala:45]: I don't like the use of blank parameters... is it possible to define it without?

File: resources/public/api/conf/P1.0/application.raml [hunk 2]
```diff
@@ -93,6 +93,11 @@ uses:
           type: string
           required: true
           example: Bearer <BEARER_TOKEN>
+        CorrelationId:
+          description: A CorreclationId provided by you.
```
Comment (owen-opencast) [inline resources/public/api/conf/P1.0/application.raml:97]: Spelling

File: app/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingController.scala (hunk 1 above)
Comment (owen-opencast) [reply app/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingController.scala:45]: i.e. withCorrelationId {
 withJsonBody { matchCitizen => ...
 }
}

**Review by owen-opencast — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingController.scala (hunk 1 above)
Comment (parkeradam) [reply app/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingController.scala:45]: This doesnt work. Unsure why, might be as theres no paramter required in the function other than the implicit, Might just be my misunderstanding of scala syntax.
Required : Function0[Future[Result]]
Found: Future[Result] 
when trying the above.

File: app/uk/gov/hmrc/individualsmatchingapi/controllers/CommonController.scala [hunk 3]
```diff
@@ -71,7 +70,7 @@ trait PrivilegedAuthentication extends AuthorisedFunctions {
   val environment: String
 
   def authPredicate(scopes: Iterable[String]): Predicate =
-    scopes.map(Enrolment(_): Predicate).reduce(_ and _)
+    scopes.map(Enrolment(_): Predicate).reduce(_ or _)
```
Comment (ajchilts) [inline app/uk/gov/hmrc/individualsmatchingapi/controllers/CommonController.scala:73]: This should already be in master but no matter 🤔

**Review by ajchilts — APPROVED** [review]
LGTM

**Review by ajchilts — APPROVED** [review]

**Pull request merged**